aiida_common_workflows/workflows/relax/vasp/generator.py

```diff
@@ -0,0 +1,145 @@
+
+        # Then some specifics. We rely on the convergence tests in order to have some
+        # common ground on the precision. Here, we test the total energy convergence to within
+        # converge.cutoff_value (100, 10 and 5 meV, respectively)
+        # This is only preliminary to illustrate the concept. In the end we also want to check how the
+        # forces converge on a relaxation event (e.g. displacement of atoms, change of volume etc.).
+        # The convergence workchain supports this, but in order to keep this simple, it is not implemented here.
+        # Also, in the end we would not be calling the convergence workchain, but some more general workchain that handles
+        # parameter suggestions, at least in a subdomain (say, periodic plane wave codes).
+        if protocol == 'fast':
+            parameters.update({'sigma': 0.2,
+                               'algo': 'V'})
+            builder.converge.cutoff_value = DataFactory('float')(0.1)
+            builder.relax.algo = DataFactory('str')('cg')
+            builder.relax.force_cutoff = DataFactory('float')(0.01)
+
+        if protocol == 'moderate':
+            parameters.update({'sigma': 0.1,
+                               'algo': 'F'})
+            builder.converge.cutoff_value =  DataFactory('float')(0.01)
+            builder.relax.algo = DataFactory('str')('cg')
+
+        if protocol == 'precise':
+            parameters.update({'sigma': 0.05,
+                               'algo': 'N'})
+            builder.converge.cutoff_value = DataFactory('float')(0.005)
+            builder.relax.algo = DataFactory('str')('rd')
+
+        # Set parameters
+        inputs.parameters = DataFactory('dict')(dict=parameters)
+
+        # Set potentials and their mapping
+        inputs.potential_family = DataFactory('str')(potential_family)
+        inputs.potential_mapping = DataFactory('dict')(dict=potential_mapping)
+
+        if relaxation_type == RelaxType.ATOMS:
+            builder.relax.positions = DataFactory('bool')(True)
+            builder.relax.shape = DataFactory('bool')(False)
+            builder.relax.volume = DataFactory('bool')(False)
+        elif relaxation_type == RelaxType.CELL:
+            builder.relax.positions = DataFactory('bool')(False)
+            builder.relax.shape = DataFactory('bool')(True)
+            builder.relax.volume = DataFactory('bool')(True)
+        elif relaxation_type == RelaxType.ATOMS_CELL:
+            builder.relax.positions = DataFactory('bool')(True)
+            builder.relax.shape = DataFactory('bool')(True)
+            builder.relax.volume = DataFactory('bool')(True)
+        else:
+            raise ValueError('relaxation type `{}` is not supported'.format(relaxation_type.value))
+
+        if threshold_forces is not None:
+            builder.relax.force_cutoff = threshold_forces
+        else:
+            raise ValueError('Please supply the cutoff for the forces.')
```

Comment:
According to [the wiki[(https://github.com/aiidateam/aiida-common-workflows/wiki/1.-Relaxation-of-a-structure) the `threshold_forces` is optional and if not defined, the protocol should define sensible value

Reply:
Indeed. Thanks. Now fixed.